app/controllers/post.py

```diff
@@ -2,21 +2,29 @@
 
 from app.instances.db import db
 from app.models.Post import Post
+from app.models.Category import Category
 from config import posts
 
 
-def create_post(title, body, tags):
+def create_post(title, body, categories):
     if g.user is None:
         return abort(403)
 
     # We don't really need to provide too much data since this is
     # validated client-side
-    if not (posts['min_title'] <= len(title) <= posts['max_title']):
+    if not posts['min_title'] <= len(title) <= posts['max_title']:
         return abort(400)
-    if not (posts['min_len'] <= len(body) <= posts['max_len']):
+    if not posts['min_len'] <= len(body) <= posts['max_len']:
         return abort(400)
+    if not len(categories) < posts['max_tags']:
+        return abort(400)
+
+    for category in categories:
+        if Category.query.filter_by(name=category).scalar() is None:
```

Comment:
Rewrite this as `if not Category.query.filter_by(name=category).first()` - `Query.scalar` raises an error if it's not found.